cmd/otel-agent/config/agent_config.go

```diff
@@ -175,8 +175,13 @@ func NewConfigComponent(ctx context.Context, ddCfg string, uris []string) (confi
 	if addr := ddc.Traces.Endpoint; addr != "" {
 		pkgconfig.Set("apm_config.apm_dd_url", addr, pkgconfigmodel.SourceFile)
 	}
-	if ddc.Traces.ComputeTopLevelBySpanKind {
-		pkgconfig.Set("apm_config.features", []string{"enable_otlp_compute_top_level_by_span_kind"}, pkgconfigmodel.SourceFile)
+
+	if pkgconfig.Get("apm_config.features") == nil {
+		apmConfigFeatures := []string{"enable_receive_resource_spans_v2"}
+		if ddc.Traces.ComputeTopLevelBySpanKind {
+			apmConfigFeatures = append(apmConfigFeatures, "enable_otlp_compute_top_level_by_span_kind")
+		}
+		pkgconfig.Set("apm_config.features", apmConfigFeatures, pkgconfigmodel.SourceDefault)
 	}
```

Comment:
```suggestion
    apmConfigFeatures := []string{"enable_receive_resource_spans_v2"}
    if ddc.Traces.ComputeTopLevelBySpanKind {
		apmConfigFeatures = append(apmConfigFeatures, "enable_otlp_compute_top_level_by_span_kind")
	}
    pkgconfig.Set("apm_config.features", apmConfigFeatures, pkgconfigmodel.SourceFile)
```

Reply:
I included the `pkgconfig.Get("apm_config.features") == nil` check because apm_config.features might be set from the datadog.yaml config file; if it is, that should be used instead of the default `apmConfigFeatures`